src/tools/compiletest/src/common.rs

```diff
@@ -79,6 +80,42 @@ string_enum! {
     }
 }
 
+impl ::std::str::FromStr for TestSuite {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "codegen-units" => Self::CodegenUnits,
+            "coverage" => Self::Coverage,
+            "coverage-run-rustdoc" => Self::CoverageRunRustdoc,
+            "crashes" => Self::Crashes,
+            "debuginfo" => Self::Debuginfo,
+            "incremental" => Self::Incremental,
+            "mir-opt" => Self::MirOpt,
+            "pretty" => Self::Pretty,
+            "run-make" => Self::RunMake,
+            "rustdoc" => Self::Rustdoc,
+            "rustdoc-gui" => Self::RustdocGui,
+            "rustdoc-js" => Self::RustdocJs,
+            "rustdoc-js-std" => Self::RustdocJsStd,
+            "rustdoc-json" => Self::RustdocJson,
+            "rustdoc-ui" => Self::RustdocUi,
+            "ui" => Self::Ui,
+            "ui-fulldeps" => Self::UiFullDeps,
+            s => {
+                if let Some((test_suite @ ("assembly" | "codegen"), codegen_backend)) =
+                    s.split_once('-')
+                    && CodegenBackend::try_from(codegen_backend).is_ok()
+                {
+                    if test_suite == "assembly" { Self::Assembly } else { Self::Codegen }
+                } else {
+                    return Err(format!("unknown `TestSuite` variant: `{s}`"));
+                }
+            }
```

Comment:
Suggestion [EXPLICIT 1/2]: can you actually do the "dumb" but obvious thing of listing them out? I.e. I would actually prefer this to be the "dumb" variants

```rs
"assembly-llvm" => Self::AssemblyLlvm,
"assembly-gcc" => Self::AssemblyGcc,
"assembly-cranelift" => Self::AssemblyCranelift,
```

and such. That way, it's obvious which test suites actually exist, and you can specifically match against one particular test suite if necessary, or exhaustively match all test suites. If you wish to refer to all such assembly test suites, I would expect that to use `TestMode::Assembly`.